README.md

```diff
@@ -33,5 +33,8 @@ A Integration test excels at describing and verifying how the application actual
 
 A Unit test focuses on one part of the application. They ignore the rest application, and describe and verify a focused part. Unit tests tend to be faster than Integration tests. That speed comes at the cost of less confidence that the application will work. The added focus lets us better describe the logic of specific parts of the code. The focus will give better errors of what is causing the test to fail.
 
+## Building and maintaining a test suite
 
+When it comes to building a project, we will want a combination of Integration and Unit focused tests. We can picture a Pyramid as what we are targeting. At the top of the Pyramid are Integration focused tests, and at the bottom are Unit focused tests. Between the top and bottom are tests that are something between a Unit and Integration test. We want a few high value Integration focused tests,. As we travel down the Pyramid, the focus should become Unit focused the further we travel.
```

Comment:
The relationship between position in the pyramid and number of tests could be expressed more clearly here; I'm not sure the students will get it first time.

Reply:
I'll elaborate in my next round of edits. Thanks for the note!

If you feel so inclined, this is the first lesson the students will read on testing. It will be followed by these repos in this exact order:

https://github.com/powerhome/phrg-rspec-intro

https://github.com/powerhome/phrg-rspec-intro-lab

https://github.com/powerhome/phrg-testing-public-api

https://github.com/powerhome/phrg-testing-intro-quiz (This is styled nicely in the learn.co UI, but looks like jibberish as markdown.)

Those complete a first round on testing, before the Rails unit.  My second round, at the moment, yet to be written, currently looks like, to be added 4 weeks later, after the Rails unit, right next to your section on CoBRA:

```
Testing: TDD (is_palindrome example?)
Testing: Fixtures lesson
Testing: FactoryBot lesson
Testing: FactoryBot lab
Testing: Timecop lesson - use example from nitro?
Testing: Timecop lab - something breaks with created_at that needs fixing
Testing: SimpleCov - Code coverage
Testing: 99 Bottles
```

Reply:
@benlangfeld I agree, I think a graphic, and some more language about this would help.

Reply:
There are a lot of graphics to choose from. Most mention Acceptance tests.  How does this graphic look:

![test-pyramid-adventures-in-qa](https://user-images.githubusercontent.com/12770108/41993856-84503270-7a1a-11e8-9279-1860f1965ac5.png)

Reply:
I would rather one just labeled Integration and Unit with a gradient of color between the two.

Reply:
<img width="738" alt="screen shot 2018-06-27 at 3 28 05 pm" src="https://user-images.githubusercontent.com/584017/41995161-c3d9970c-7a1e-11e8-9d32-649295ea6f21.png">
A few integration tests, and lot of more unit focused tests.

Reply:
This is from Martin Fowler. I like it, but don't want to confuse and distract people with the service section of the graphic.

<img src="https://martinfowler.com/bliki/images/testPyramid/test-pyramid.png">

Reply:
![68747470733a2f2f6d617274696e666f776c65722e636f6d2f626c696b692f696d616765732f74657374507972616d69642f746573742d707972616d69642e706e67](https://user-images.githubusercontent.com/210221/41996555-a1824822-7a2b-11e8-84cd-5d834446554a.png)

Reply:
Lolz. Thank @benlangfeld, but these I think copy right law would make that prohibited.

Reply:
How well can you draw bunnies and tortoises that look something like, but not too much like, the above graphic @garettarrowood ? 🤣